### ISSUE TITLE
name.com failing test: 20:_Internationalized_name

### ISSUE THREAD

**Issue by tlimoncelli** [StackExchange/dnscontrol#168]
```
        --- FAIL: TestDNSProviders/stackosphere.com/20:_Internationalized_name (1.13s)
        	integration_test.go:131: CREATE A ööö.stackosphere.com 1.2.3.4 ttl=300
        	integration_test.go:144: Expected 0 corrections on second run, but found 2.
        	integration_test.go:146: #0: DELETE A xn--ndaaa.stackosphere.com 1.2.3.4 ttl=300
        	integration_test.go:146: #1: CREATE A ööö.stackosphere.com 1.2.3.4 ttl=300
```

**Comment by pmoroney** [comment]
Yeah, we return all international names in their punycode format. I can modify the provider to convert to unicode if that is what we want. My only worry is would that create issues going the other direction? e.g. I have xn--ndaaa.stackosphere.com in my dnsconfig.js, would it want to delete ööö.stackosphere.com to create the punycode version?

Actually, I think a more intelligent fix would be for the function that compares domain names to be punycode/unicode agnostic. I'll look into this.

**Comment by captncraig** [comment]
By convention we want all unicode names in dnsconfig.js. I am now considering a validation check to disallow punycode in the config.

**Comment by captncraig** [comment]
There is a function you can call at the beginning of the provider function to punycode all names in the domain. Many providers do this. See https://github.com/StackExchange/dnscontrol/blob/d55b20ecdbcf61176764a426695415545c59da10/providers/gandi/gandiProvider.go#L63

**Comment by pmoroney** [comment]
Awesome, I'll add that.

Can I get access to the code for running that test so I can verify it passes?

Also I noticed that the `DomainConfig.Punycode()` function doesn't operate on ALIAS or SRV record targets. Should I create a new PR for that?

**Comment by tlimoncelli** [comment]
I'm working on a PR that fixes DomainConfig.Punycode() on ALIAS and SRV records.

You can run the tests with your own API keys this way:

```
export NAMEDOTCOM_DOMAIN='CHANGE_TO_A_TEST_DOMAIN_NAME'
export NAMEDOTCOM_URL='https://api.name.com/api'
export NAMEDOTCOM_KEY='CHANGE_THIS_TO_YOUR_API_KEY'
export NAMEDOTCOM_USER='CHANGE_THIS_TO_YOUR_API_USER'
cd dnscontrol/integrationTest
go test -v -verbose -provider NAMEDOTCOM
```

(Sadly we don't have a set of API user/keys that we can hand out to people.)

**Comment by pmoroney** [comment]
Awesome, Thanks!
I can look into providing some Name.com credentials for our OT&E environment to this project.

**Comment by tlimoncelli** [comment]
I don't use any Punycode domains, so I'm a little ignorant here:

> By convention we want all unicode names in dnsconfig.js. I am now
> considering a validation check to disallow punycode in the config.

I agree that dnsconfig.js should have 1 way to do things (in this case, always use Unicode). However, might we want to add a function so that people can enter punycode when they want:

    A(pc("xn--ndaaa"), "1.2.3.4"),

The pc() function could do validation, etc.

**Comment by tlimoncelli** [comment]
> I can look into providing some Name.com credentials for our OT&E environment to this project.

I have a set of OT&E credentials  but the tests fail badly when I use them.  We've been using a separate  account.  If you could help me get the OT&E creds working, I'd appreciate it!

**Comment by pmoroney** [comment]
I just created the following PR: https://github.com/StackExchange/dnscontrol/pull/170

I also got it so the OT&E credentials should work now. Let me know if there are any other issues.

**Comment by tlimoncelli** [comment]
All fixed as of 93764da4e3b39653ba2a20dfb2026c69ba1c1f2d. Thanks, @pmoroney!
